src/app/core/session/session-service/session-manager.service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ *     This file is part of ndb-core.
+ *
+ *     ndb-core is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     ndb-core is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Injectable } from "@angular/core";
+
+import { LocalSession } from "./local-session";
+import { AuthUser } from "../auth/auth-user";
+import { SyncService } from "../../database/sync.service";
+import { UserService } from "../../user/user.service";
+import { LoginStateSubject, SessionType } from "../session-type";
+import { LoginState } from "../session-states/login-state.enum";
+import { Router } from "@angular/router";
+import { KeycloakAuthService } from "../auth/keycloak/keycloak-auth.service";
+import { LocalAuthService } from "../auth/local/local-auth.service";
+import { environment } from "../../../../environments/environment";
+
+/**
+ * A synced session creates and manages a LocalSession and a RemoteSession
+ * and handles the setup of synchronisation.
+ *
+ * also see
+ * [Session Handling, Authentication & Synchronisation]{@link /additional-documentation/concepts/session-and-authentication-system.html}
+ */
+@Injectable()
+export class SessionManagerService {
+  private remoteLoggedIn = false;
+  constructor(
+    private remoteAuthService: KeycloakAuthService,
+    private localAuthService: LocalAuthService,
+    private localSession: LocalSession,
+    private syncService: SyncService,
+    private userService: UserService,
+    private loginStateSubject: LoginStateSubject,
+    private router: Router,
+  ) {}
+
+  remoteLogin() {
+    this.remoteAuthService.authenticate();
+  }
+
+  offlineLogin() {
+    const user = this.localAuthService.login();
+    return this.initialiseUser(user);
+  }
+
+  canLoginOffline(): boolean {
+    return this.localAuthService.canLoginOffline();
+  }
+
+  logout() {
+    if (this.remoteLoggedIn) {
+      this.remoteAuthService.logout();
+    } else {
+      this.userService.user = undefined;
+      this.loginStateSubject.next(LoginState.LOGGED_OUT);
```

Comment:
shouldn't loginState etc. also reset for remote logout? I guess this does happen somewhere else already but for cleaner code, is there any chance to do this here, in the same place?

Reply:
a similar point for `remoteLogin()`: from the service here, it's not really clear how/when `initializeUser()` is called.

Reply:
Improve function names and add comments for explanation